app/assets/javascripts/app/azlist.js

```diff
@@ -0,0 +1,27 @@
+
+var oldText = "All Facilities"
+
+function removeExtraClasses(classlist) {
+    if (classlist.includes("all")) return "all"
+    else if(classlist.includes("recent")) return "recent"
+    return "js--azlist" + classlist.replace(" ", "").split("js--azlist")[1]
+}
+
+$(document).ready(function () {
+    oldText = $(".allHeader").text()
+})
+$(document).on("click", ".js--az_listing", function () {
+    $(".js--facility_listing").hide();
+    var category = removeExtraClasses( $(this).attr("class") )
```

Comment:
I could split at the beginning and then do the check, but two of the cases don't require it. I simply added spaces to pad the string to prevent this condition. If you prefer it for readability, then it's not a big deal.